spec/controllers/projects_controller_spec.rb

```diff
@@ -1,386 +1,350 @@
+            subject
+            expect(assigns(:projects)).to eq(projects)
+          end
+
+          it 'renders the destroy template' do
+            allow(student_project).to receive(:destroy)
+            allow(controller).to receive(:latest_projects).and_return(projects)
+            expect(subject).to render_template(:destroy)
+          end
+        end
+
+        describe "cannot destroy other user's project" do
+          let(:request) {
+            delete :destroy, params: {
+              lesson_id: lesson.slug, id: admin_project.id
+            }, xhr: true
+          }
+
+          before do
+            stub_current_user_with(student)
+            stub_lesson
+            stub_project(admin_project)
+            stub_authorize_to_raise(:destroy, admin_project)
+          end
+
+          it_behaves_like 'unauthorized XHR request'
+        end
+      end
+
+      context 'admin user' do
+        describe 'can destroy his/her project' do
+          let(:request) {
+            delete :destroy,
+                   params: { lesson_id: lesson.slug, id: admin_project.id },
+                   xhr: true
+          }
+
+          before do
+            stub_current_user_with(admin)
+            stub_lesson
+            stub_project(admin_project)
+            stub_authorize(:destroy, admin_project)
+            allow(controller).to receive(:new_project).and_return(admin_project)
+            allow(controller).to receive(:latest_projects).and_return(projects)
+          end
+
+          it 'assigns recent submissions' do
+            allow(admin_project).to receive(:destroy)
+            request
+            expect(assigns(:projects)).to eq(projects)
+          end
+
+          it 'renders the destroy template' do
+            allow(admin_project).to receive(:destroy)
+            request
+            expect(response).to render_template(:destroy)
+          end
+        end
+
+        describe "can destroy other user's project" do
+          subject {
+            delete :destroy,
+                   params: { lesson_id: lesson.slug, id: student_project.id },
+                   xhr: true
+          }
+
+          before do
+            stub_current_user_with(admin)
+            stub_lesson
+            stub_project(student_project)
+            stub_authorize(:destroy, student_project)
+            allow(controller).to receive(:new_project).and_return(admin_project)
+            allow(controller).to receive(:latest_projects).and_return(projects)
+          end
+
+          it 'assigns recent submissions' do
+            allow(student_project).to receive(:destroy)
+            subject
+            expect(assigns(:projects)).to eq(projects)
+          end
+
+          it 'renders the destroy template' do
+            allow(student_project).to receive(:destroy)
+            expect(subject).to render_template(:destroy)
+          end
+        end
+      end
+    end
+  end
+
+  def stub_current_user_with(user)
```

Comment:
These methods for stubbing aren't needed, they make the tests borderline unreadable imo. 

You can define a before do in the top level describe for this test and then override the stubs we need to change for each method in before do blocks within the method describe scopes.

This is the way we do it in every other test in the app.